src/components/Router/index.js

```diff
@@ -82,105 +103,40 @@ class DesktopRouter extends Component {
-    return (
-      this.state.mobileConnected ? <p>Mobile connected</p> :
-        <StepsRouter {...props} onStepChange={this.onStepChange} mobileUrl={mobileUrl}/>
-    )
-  }
-}
-
-
-class StepsRouter extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      step: this.props.step || 0,
-      componentsList: this.createComponentListFromProps(this.props),
-    }
-    this.unlisten = history.listen(({state = this.initialState}) => {
-      this.setState(state)
-    })
-  }
-
-  nextStep = () => {
-    const components = this.state.componentsList
-    const currentStep = this.state.step
-    const newStepIndex = currentStep + 1
-    if (components.length === newStepIndex){
-      events.emit('complete')
-    }
-    else {
-      this.setStepIndex(newStepIndex)
+  buildComponentsList = () => {
+    const params = {
+      flow: this.state.flow,
+      documentType: this.props.documentType,
+      steps: this.props.options.steps
     }
+    return componentsList(params)
   }
 
-  previousStep = () => {
-    const currentStep = this.state.step
-    this.setStepIndex(currentStep - 1)
-  }
-
-  setStepIndex = (newStepIndex) => {
-    const state = { step: newStepIndex }
-    const path = `${location.pathname}${location.search}${location.hash}`
-    this.props.onStepChange && this.props.onStepChange(newStepIndex)
-    history.push(path, state)
-  }
-
-  trackScreen = (screenNameHierarchy, properties = {}) => {
-    const { step } = this.currentComponent()
-    sendScreen(
-      [step.type, ...wrapArray(screenNameHierarchy)],
-      {...properties, ...step.options})
-  }
-
-  currentComponent = () => this.state.componentsList[this.state.step]
-
-  componentWillReceiveProps(nextProps) {
-    const componentsList = this.createComponentListFromProps(nextProps)
-    this.setState({componentsList})
-  }
-
-  componentWillMount () {
-    this.setStepIndex(this.state.step)
-  }
-
-  componentWillUnmount () {
-    this.unlisten()
-  }
-
-  createComponentListFromProps = ({documentType, options:{steps}}) =>
-    createComponentList(steps, documentType)
-
-  render = ({options: {...globalUserOptions}, ...otherProps}) => {
-    const componentBlob = this.currentComponent()
-    const CurrentComponent = componentBlob.component
+  render = (props) => {
+    const components = this.buildComponentsList()
     return (
-      <div>
-        <CurrentComponent
-          {...{...componentBlob.step.options, ...globalUserOptions, ...otherProps}}
-          nextStep = {this.nextStep}
-          previousStep = {this.previousStep}
-          trackScreen = {this.trackScreen}/>
-      </div>
+      <StepsRouter {...props}
+        componentsList={components}
+        flow={this.state.flow}
+        nextFlow={this.nextFlow}
+        step={this.state.step}
+        onStepChange={this.onStepChange}
+        roomId={this.state.roomId}
+        isMobileFlow={props.options.mobileFlow}
```

Comment:
`isMobileFlow` is always `false` for `DesktopRouter` and `true` for `MobileRouter`. By reading those two components in isolation one can wonder if that might not be the case.
I would recommend setting `isMobileFlow` in the `Router`, so these two children (mobile and desktop) who shouldn't control that value just pass it down.

Reply:
working on this as part of https://github.com/onfido/onfido-sdk-ui/pull/217